app-config-electron/src/index.ts

```diff
@@ -0,0 +1,27 @@
+import type { WebPreferences } from 'electron';
+import type { ExportedConfig } from '@app-config/main';
+import * as path from 'path';
+
+export function addAppConfigPreload(config: ExportedConfig, baseWebPreferences?: WebPreferences) {
+  let webPreferences: WebPreferences = {};
+
+  if (baseWebPreferences) {
+    webPreferences = baseWebPreferences;
+  }
+
+  const userPreload = baseWebPreferences?.preload;
+  const userArguments = baseWebPreferences?.additionalArguments;
+
+  const preloadArguments = userArguments || [];
+
+  preloadArguments.push(`--app-config=${JSON.stringify(config)}`);
+
+  if (userPreload) {
+    preloadArguments.push(`--user-preload=${userPreload}`);
+  }
+
+  webPreferences.preload = path.join(__dirname, 'preloader.js');
```

Comment:
I wonder if we could use `path.resolve` without `__dirname`? This is deprecated in Node.js / won't work when migrating to ES modules.

Reply:
`path.resolve` seems to go to the wrong path. It uses the current project path and not the path where that file is actually located. (/home/john/dev/app-config/examples/preload.js not /home/john/dev/app-config/app-config-electron/dist/preloader.js)

`import.meta.url` seems to be the replacement for `__dirname`, but it's not usable without using ES modules. https://nodejs.org/dist/latest/docs/api/esm.html#esm_import_meta_url

Reply:
Double checking, what about `require.resolve`? Not that that's really any better

Reply:
`require.resolve` works, but I think that has the same problem as `__dirname` right?

Reply:
Yup

Reply:
Did we want to do anything about this, or is it fine for now?

Reply:
All good